lib/runner/extensions/event.command.js

```diff
@@ -308,19 +308,24 @@ module.exports = {
 
                     // if it is defined that certain variables are to be synced back to result, we do the same
                     track && result && track.forEach(function (variable) {
-                        if (!(_.isObject(result[variable]) && payload.context[variable])) { return; }
-
-                        // ensure that variablescope is treated accordingly
-                        // @todo find a better way to not sync entire scope, but receive changes and update the existing
-                        // scope using that
-                        if (_.isFunction(payload.context[variable].syncVariablesFrom)) {
-                            payload.context[variable].clear();
-                            _.forEach(result[variable].values, function (def) {
-                                payload.context[variable].values.add(def);
-                            });
+                        var mutations,
+                            contextVariable = payload.context[variable];
+
+                        if (!(_.isObject(result[variable]) && result[variable].mutations && contextVariable)) {
```

Comment:
The `if` block has to be shifted to before the `var ` above, purely for the reasons of readability.